eva-accession-import/src/main/java/uk/ac/ebi/eva/accession/dbsnp/io/DbsnpJsonItemReader.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2018 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.ac.ebi.eva.accession.dbsnp.io;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.batch.item.ItemReader;
+import uk.ac.ebi.eva.accession.dbsnp.model.DbsnpJson;
+import uk.ac.ebi.eva.accession.dbsnp.parameters.InputParameters;
+
+import java.io.File;
+import java.io.IOException;
+
+/**
+ * Item reader for Dbsnp Json import job
+ */
+public class DbsnpJsonItemReader implements ItemReader<DbsnpJson> {
+
+    private static final Logger logger = LoggerFactory.getLogger(DbsnpJsonItemReader.class);
+
+    private InputParameters parameters;
+
+    public DbsnpJsonItemReader(InputParameters parameters) {
+        this.parameters = parameters;
+    }
+
+    @Override
+    public DbsnpJson read() throws Exception {
+        JsonFactory jsonFactory = new JsonFactory();
+        JsonParser jp = jsonFactory.createJsonParser(new File(parameters.getDbsnp()));
+        jp.setCodec(new ObjectMapper());
+        JsonNode jsonNode = jp.readValueAsTree();
+        return parseDbsnpJsonPlacements(jsonNode);
+    }
+
+    /**
+     * Parses the JSON as per https://github.com/ncbi/dbsnp/blob/master/tutorials/hadoop_json_placement.py
+     * @param JsonNode jsonRootNode
+     * @return parsed DbsnpJson object
+     */
+    private DbsnpJson parseDbsnpJsonPlacements(JsonNode jsonRootNode) throws IOException {
```

Comment:
This does not seem to deal with multi-allelic variants (ex: A/T,G). Either 

1. DbsnpJson has to be modified to accommodate multiple alternates or 
2. This function should return multiple DbsnpJson objects. 

I think the former is easier from an implementation standpoint. @cyenyxe Any thoughts?

Reply:
If `DbsnpJson` supports multiple alternates, they will need to be split later on when we construct the final objects to be serialized. But it is true that it may be easier to deal with that split at that point instead of during the reading, where the input is quite convoluted.